### PR TITLE
Adding cuda-cuobjdump, cuda-cuxxfilt, and cuda-nvprune recipes

### PR DESCRIPTION
**Pull request by adibbley**
Adding cuda-cuobjdump, cuda-cuxxfilt, and cuda-nvprune recipes

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by adibbley** [comment]
cc @jakirkham

File: recipes/cuda-cuobjdump/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  sha256: 00688ad84b04df7e6a39e6d4d5fb29fda42d36bdf7e3a432c332ebafbed6fb90  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
```
Comment (bdice) [inline recipes/cuda-cuobjdump/meta.yaml:28]: Per https://github.com/conda-forge/cuda-version-feedstock/issues/1#issuecomment-1462998398, we should pin to `cuda-version`.
```suggestion
  build:
    - {{ compiler("c") }}
    - {{ compiler("cxx") }}
  run:
    - cuda-version =12.0
```

File: recipes/cuda-cuxxfilt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  sha256: 0d6276d4a49e7e394b7f8ee150bb6d538297334c7d6b3d1266fa29311ba933f4  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
```
Comment (bdice) [inline recipes/cuda-cuxxfilt/meta.yaml:28]:
```suggestion
  build:
    - {{ compiler("c") }}
    - {{ compiler("cxx") }}
  run:
    - cuda-version =12.0
```

**Review by bdice — COMMENTED** [review]
Suggested cuda-version pinnings.

File: recipes/cuda-cuobjdump/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{% set name = "cuda-cuobjdump" %}
+{% set version = "12.0.76" %}
```
Comment (bdice) [inline recipes/cuda-cuobjdump/meta.yaml:2]:
```suggestion
{% set version = "12.0.76" %}
{% set cuda_version = "12.0" %}
```

File: recipes/cuda-cuobjdump/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+build:
+  number: 0
+  skip: true  # [osx]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  run:
+    - cuda-version =12.0
```
Comment (bdice) [inline recipes/cuda-cuobjdump/meta.yaml:30]:
```suggestion
    - cuda-version {{ cuda_version }}
```

File: recipes/cuda-cuxxfilt/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+{% set name = "cuda-cuxxfilt" %}
+{% set version = "12.0.76" %}
```
Comment (bdice) [inline recipes/cuda-cuxxfilt/meta.yaml:2]:
```suggestion
{% set version = "12.0.76" %}
{% set cuda_version = "12.0" %}
```

File: recipes/cuda-cuxxfilt/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+build:
+  number: 0
+  skip: true  # [osx]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  run:
+    - cuda-version =12.0
```
Comment (bdice) [inline recipes/cuda-cuxxfilt/meta.yaml:30]:
```suggestion
    - cuda-version {{ cuda_version }}
```

File: recipes/cuda-nvprune/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{% set name = "cuda-nvprune" %}
+{% set version = "12.0.76" %}
```
Comment (bdice) [inline recipes/cuda-nvprune/meta.yaml:2]:
```suggestion
{% set version = "12.0.76" %}
{% set cuda_version = "12.0" %}
```

File: recipes/cuda-nvprune/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+build:
+  number: 0
+  skip: true  # [osx]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  run:
+    - cuda-version =12.0
```
Comment (bdice) [inline recipes/cuda-nvprune/meta.yaml:30]:
```suggestion
    - cuda-version {{ cuda_version }}
```

**Review by bdice — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏 

Let's get this into feedstocks. We can follow up on anything else there 🙂

**Pull request merged**